### DISCUSSION TITLE
Removing certain entries from legends

### DISCUSSION THREAD

**Discussion by dreese80** [Samreay/ChainConsumer#124] [Ideas]
We noticed that chainconsumer automatically adds an entry to the legend for each chain that is added.  In some cases, it is useful not to add an entry to the legend (for example if you add markers within the triangle plot).  We added a field to chain to decide whether the name should be visible or not in the legend (see previous discussion on weighted chains - the attachment contains an example of how we did this).  Perhaps, though, this is not the best way of doing this.  Please let us know if we should proceed differently.

**Comment by Samreay** [comment]
Hi Daniel, this seems smart to me, I'll add that kwarg to make show on a legend into the code today :)

**Reply by dreese80** [reply]
Thanks!

**Comment by Samreay** [comment]
Hey @dreese80, `show_label_in_legend` should now be available in 1.1.0

**Reply by dreese80** [reply]
Hi Samuel, my intern and I have tested the show_label_in_legend option.  It works if you explicitly create a chain. But if you create one indirectly, for example through the add_marker function, it doesn't recognise the show_label_in_legend option.  A workaround consists in directly accessing the chain created by add_marker through the _chains dictionary within a ChainConsumer object and manually setting the field show_label_in_legend to the desired value, but it may be better to add the option directly in add_marker and other similar functions (or perhaps by adding a **kwargs argument to these functions).

**Reply by Samreay** [reply]
Aaaah good catch - please try now with version 1.1.1 of ChainConsumer :)

**Reply by dreese80** [reply]
We tested it and it works now. Thanks!
